test/cli/and-and-always-truthy/test.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+extend T::Sig
+
+sig {returns(TrueClass)}
+def always_true; true; end
+
+sig {returns(FalseClass)}
+def always_false; false; end
+
+def example1
+  if always_true && T.unsafe(true)
+    puts 'here'
+  end
+end
+
+def example2
+  if always_false && T.unsafe(true)
+  end
+end
+
+sig {params(always_true: TrueClass).void}
+def example3(always_true)
+  # TODO: We don't get the same sort of nice error message in this case.
+  if always_true && T.unsafe(true)
+    puts 'here'
+  end
+end
```

Comment:
I'm fine with this limitation for the time being, because I think that something is better than nothing.

Even better might be to desugar this to have some sort of `<Magic>.<and-and-else>(always_true)` call in it:

```ruby
if always_true
  T.unsafe(true)
else
  <Magic>.<and-and-else>(always_true)
end
```

but then the first dead instruction in `inference.cc` would still be the `always_true` ident and the call would be the *second* instruction, and this would make trees larger, etc. etc.

This could happen in reasonable looking code, too:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/5544532/182268947-61951c94-d0a8-4552-85ea-8895bffa7a84.png">

but `¯\_(ツ)_/¯`